### ISSUE TITLE
[ENH]: matplotlib 3.12 prelease wheels 

### ISSUE THREAD

**Issue by jarrodmillman** [matplotlib/matplotlib#26736]
### Problem

Now that #26647 is merged, would it be possible to have a matplotlib 3.8.0rc2 release with 3.12 wheels soon. It would be helpful for folks (like me) trying to test downstream (from matplotlib) packages against 3.12.

**Comment by greglucas** [comment]
I would be bold as to say we should just release 3.8 without another rc? @ksunden do you have an expected release date for either of these.

**Comment by jarrodmillman** [comment]
Also, I noticed that when you updated the version number after the  3.8.0rc1 release, you used `3.8.0.dev`. That is less than the older release candidate. So if you try to install matplotlib with
```
$ pip install --pre --upgrade --extra-index-url https://pypi.anaconda.org/scientific-python-nightly-wheels/simple matplotlib
Looking in indexes: https://pypi.org/simple, https://pypi.anaconda.org/scientific-python-nightly-wheels/simple
Collecting matplotlib
  Using cached matplotlib-3.8.0rc1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (11.6 MB)
Collecting contourpy>=1.0.1
  Using cached contourpy-1.1.1rc1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (302 kB)
Collecting cycler>=0.10
  Using cached cycler-0.11.0-py3-none-any.whl (6.4 kB)
Collecting fonttools>=4.22.0
  Using cached fonttools-4.42.1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.9 MB)
Collecting kiwisolver>=1.0.1
  Using cached kiwisolver-1.4.5-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.4 MB)
Collecting numpy>=1.21
  Downloading https://pypi.anaconda.org/scientific-python-nightly-wheels/simple/numpy/2.0.0.dev0/numpy-2.0.0.dev0-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (18.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 18.3/18.3 MB 16.6 MB/s eta 0:00:00
Collecting packaging>=20.0
  Using cached packaging-23.1-py3-none-any.whl (48 kB)
Collecting pillow>=6.2.0
  Using cached Pillow-10.0.0-cp311-cp311-manylinux_2_28_x86_64.whl (3.4 MB)
Collecting pyparsing>=2.3.1
  Using cached pyparsing-3.1.1-py3-none-any.whl (103 kB)
Collecting python-dateutil>=2.7
  Using cached python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
Collecting numpy>=1.21
  Using cached numpy-1.26.0rc1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (18.2 MB)
Collecting six>=1.5
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Installing collected packages: six, pyparsing, pillow, packaging, numpy, kiwisolver, fonttools, cycler, python-dateutil, contourpy, matplotlib
Successfully installed contourpy-1.1.1rc1 cycler-0.11.0 fonttools-4.42.1 kiwisolver-1.4.5 matplotlib-3.8.0rc1 numpy-1.26.0rc1 packaging-23.1 pillow-10.0.0 pyparsing-3.1.1 python-dateutil-2.8.2 six-1.16.0
```

**Comment by jarrodmillman** [comment]
```
>>> from packaging import version
>>> version.parse("3.8.0rc1") < version.parse("3.8.0.dev1951+g5f785e3e19")
False
>>> version.parse("3.8.0rc1") < version.parse("3.8.0rc1.dev1951+g5f785e3e19")
False
>>> version.parse("3.8.0rc1") < version.parse("3.8.0rc1.dev0")
False
>>> version.parse("3.8.0rc1") < version.parse("3.8.0rc2.dev0")
True
>>> version.parse("3.8.0rc1") < version.parse("3.8.0rc2.dev1951+g5f785e3e19")
True
```

So after releasing `3.8.0rc1` the basic version number should be updated to `3.8.0rc2.dev0` or `3.8.0rc2.dev`.

**Comment by ksunden** [comment]
The version number precedence things is because we have not merged the rc's tag into main, which as I understand is always how we have done it, waiting for final before doing that merge process. See also astropy/astropy#15284

Yes, I intend to do a final release of 3.8.0 this week.

I was to a certain degree waiting for pydata/xarray#8030 which I think has now gotten everything mpl-related sorted, though has not been merged yet, but I feel less bad about releasing when they have their fixes ready if not fully in place.

I also would tend to want 3.7.3 out prior, but I suppose there is no _real_ requirement of that (@QuLogic opinions?). I think that is ready to go, the milestone is even closed, but it has not been tagged.

**Comment by QuLogic** [comment]
Well, 3.7.3 is out now, so you can feel free to release 3.8.0 whenever.

**Comment by QuLogic** [comment]
> Now that #26647 is merged, would it be possible to have a matplotlib 3.8.0rc2 release with 3.12 wheels soon. It would be helpful for folks (like me) trying to test downstream (from matplotlib) packages against 3.12.

Do you need Matplotlib 3.8.0(rc) wheels or Python 3.12 wheels? The latter is already covered by Matplotlib 3.7.3.

**Comment by jarrodmillman** [comment]
Matplotlib 3.8.0 Python 3.12 wheels would be the most convenient. We are using `--pre` to get numpy 1.26.0rc1 in our CI, but that means we also get matplotlib 3.8.0rc1 (and then have to wait for it to build Python 3.12 wheels).

Are you still planning to release matplotlib 3.8 this week? If so, we will probably just wait until that happens before merging https://github.com/PyWavelets/pywt/pull/683 and trying to get pywavelets 1.5 out, which is needed before we can release scikit-image 0.22.

**Comment by jarrodmillman** [comment]
- https://pypi.org/project/matplotlib/3.7.3/
- https://pypi.org/project/matplotlib/3.8.0/

Thanks!!
